src/index.d.ts

```diff
@@ -0,0 +1,298 @@
+// Type definitions for Paylike
+
+import * as Promise from 'bluebird'
+
+declare class Paylike {
+  constructor(appKey: string)
+
+  transactions: Paylike.Transactions
+  merchants: Paylike.Merchants
+  cards: Paylike.Cards
+  apps: Paylike.Apps
+  service: Paylike.Service
+}
+
+declare namespace Paylike {
+
+  export interface PaylikeError extends Error {
+    message: string
+  }
+  export interface PaylikeErrorConstructor extends Error {
+    new (message: string): PaylikeError
+    (message: string): PaylikeError
+    prototype: PaylikeError
+  }
+  export const Error: PaylikeErrorConstructor
+
+
+  export interface AuthorizationError extends PaylikeError {
+    message: string
+  }
+  export interface AuthorizationErrorConstructor extends PaylikeError {
+    new (message: string): AuthorizationError
+    (message: string): AuthorizationError
+    prototype: AuthorizationError
+  }
+  export const AuthorizationError: AuthorizationErrorConstructor
+
+
+  export interface PermissionsError extends PaylikeError {
+    message: string
+  }
+  export interface PermissionsErrorConstructor extends PaylikeError {
+    new (message: string): PermissionsError
+    (message: string): PermissionsError
+    prototype: PermissionsError
+  }
+  export const PermissionsError: PermissionsErrorConstructor
+
+
+  export interface NotFoundError extends PaylikeError {
+    message: string
+  }
+  export interface NotFoundErrorConstructor extends PaylikeError {
+    new (message: string): NotFoundError
+    (message: string): NotFoundError
+    prototype: NotFoundError
+  }
+  export const NotFoundError: NotFoundErrorConstructor
+
+
+  export interface ConflictError extends PaylikeError {
+    message: string
+  }
+  export interface ConflictErrorConstructor extends PaylikeError {
+    new (message: string): ConflictError
+    (message: string): ConflictError
+    prototype: ConflictError
+  }
+  export const ConflictError: ConflictErrorConstructor
+
+
+  export interface ValidationError extends PaylikeError {
+    message: string
+  }
+  export interface ValidationErrorConstructor extends PaylikeError {
+    new (message: string): ValidationError
+    (message: string): ValidationError
+    prototype: ValidationError
+  }
+  export const ValidationError: ValidationErrorConstructor
+
+  interface Cursor<T> {
+    after: (id: string) => Cursor<T>
+    before: (id: string) => Cursor<T>
+    limit: (limit: number) => Cursor<T>
+    stream: (highWaterMark: number) => any
+    toArray: () => Promise<Array<T>>
```

Comment:
Yes they and easily be added. I'll leave that as an exercise for you ;) 
Unfortunately there are typings for pull-stream, that's why `stream` returns `any` right now.